python/plugins/processing/gui/AlgorithmDialog.py

```diff
@@ -111,6 +111,8 @@ def getParameterValues(self):
                         dest_project = QgsProject.instance()
 
                 value = self.mainWidget().outputWidgets[param.name()].getValue()
+                if value == '':
+                    value = self.parameter.generateTemporaryDestination()
```

Comment:
When does this occur? I'm concerned it may break the "skip output" option for optional outputs.